src/test/fuzz/deserialize.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "test/fuzz/deserialize.h"
+#include "test/TestUtils.h"
+#include "test/TxTests.h"
+#include "test/fuzz/FuzzTransaction.h"
+#include "test/test.h"
+#include "util/Logging.h"
+
+using namespace stellar;
+
+TestAccount
+mapToAccount(const uint8_t& index, std::vector<TestAccount> accounts)
+{
+    // Mapping accounts, is done by taking the mod of the last byte of the
+    // AccountID with the pre-generated accounts collection size, and
+    // returningthe account at that index.
+    auto end_byte = static_cast<unsigned int>(index);
+    return accounts.at(end_byte % accounts.size());
+}
+
+namespace stellar
+{
+
+TransactionFramePtr
+deserializeFuzzTransactionInput(FuzzTransactionInput const& inp,
```

Comment:
I don't think you need this machinery:
if you generate test accounts that have a "public key" 000000XX, then the mapping happens implicitly and you don't have to scan the XDR to search/replace

Reply:
Ok, I see... at first I considered the impossibility of predictability generating ED25519 keypairs, but then remembered that we aren't checking signatures here, so secret keys don't matter!